hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/common/ChecksumByteBufferImpl.java

```diff
@@ -69,7 +87,15 @@ public void update(ByteBuffer buffer) {
     if (buffer.hasArray()) {
       checksum.update(buffer.array(), buffer.position() + buffer.arrayOffset(),
           buffer.remaining());
+    } else if (BYTE_BUFFER_UPDATE != null && buffer.isDirect()) {
```

Comment:
shouldn't it check this condition first, so that the JDK9 Checksum.update(ByteBuffer) is preferred?

Reply:
Good idea, thanks.